### PR TITLE
SMP-2100: Add tmpl function for smtp secrets

### PR DESCRIPTION
**Pull request by saitejavn-harness**
1. Added tmpl func to support external secrets for SMTP
2. Fixed precedence order of "protocol" field of timescaledb 

Testing:
image

### CODE REVIEW

File: src/common/templates/_databaseconnectionsv2.tpl [hunk 1]
```diff
@@ -89,10 +89,14 @@ USAGE:
     {{- $host := include "harnesscommon.dbconnectionv2.timescaleHost" (dict "context" .context ) }}
     {{- $port := include "harnesscommon.dbconnectionv2.timescalePort" (dict "context" .context ) }}
     {{- $connectionString := "" }}
-    {{- $protocol := (include "harnesscommon.precedence.getValueFromKey" (dict "ctx" $ "keys" (list ".Values.global.database.timescaledb.protocol" ".Values.timescaledb.protocol"))) }}
+    {{- $protocol := "" }}
     {{- if not (empty .protocol) }}
         {{- $protocol = (printf "%s://" .protocol) }}
     {{- end }}
+    {{- $protocolVar := (include "harnesscommon.precedence.getValueFromKey" (dict "ctx" $ "keys" (list ".Values.global.database.timescaledb.protocol" ".Values.timescaledb.protocol"))) }}
```
Comment (Vbhatia20) [inline src/common/templates/_databaseconnectionsv2.tpl:96]: why this change?

**Review by Vbhatia20 — APPROVED** [review]

File: src/common/templates/_databaseconnectionsv2.tpl (hunk 1 above)
Comment (saitejavn-harness) [reply src/common/templates/_databaseconnectionsv2.tpl:96]: It is for making sure we set default value for "protocol" only when user didn't provide it as input in values file.
